File: apps/web/components/profile/ripples/ripple-strokes.ts
// Ripple stroke geometry, hand-ported from the SVG <path d="…"> definitions in
// issue #442 (authored against a 44×44 viewBox), mirroring the iOS
// `RippleStrokes.swift`. Ordered outermost (id 6) → innermost (id 1) so inner
// rings paint on top. `opacity` reproduces the iOS RippleBadge layering.

export type RippleStrokeDef = {
  id: number
  d: string
  opacity: number
}

export const RIPPLE_STROKES: RippleStrokeDef[] = [
  {
    id: 6,
    d: "M15.4023 2.71506 C26.241 -0.507724 41.9999 7.38652 41.9999 21.8993",
    opacity: 0.33,
  },
  {
    id: 5,
    d: "M7.37405 37.1175 C-1.10595 29.2114 0.748869 9.24398 10.831 4.78223",
    opacity: 0.33,
  },
  {
    id: 4,
    d: "M41.458 26.9565 C39.2185 38.9628 23.9232 45.4638 11.4043 40.1005",
    opacity: 0.33,
  },
  {
    id: 3,
    d: "M36.6088 19.5146 C39.4844 34.5339 18.0179 42.6778 10 31.0941",
    opacity: 0.66,
  },
  {
    id: 2,
    d: "M34.1755 13.5272 C25.9708 1.34962 4.58761 9.44313 7.58572 25.087",
    opacity: 0.66,
  },
  {
    id: 1,
    d: "M25.4147 30.7822 C22.7365 31.9714 19.5086 31.8785 16.4741 29.5504 C6.9764 22.2636 18.3687 7.65428 27.8664 14.941 C32.662 18.6202 32.1318 24.1663 29.2088 27.818",
    opacity: 1,
  },
]

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/shared/ripples/RippleStrokes.kt
package app.pbbls.android.features.shared.ripples

import androidx.compose.ui.graphics.Path

/**
 * The six Ripple ring strokes, hand-ported from the SVG `<path d="…">`
 * definitions in issue #442 (coordinates transcribed 1:1 from iOS
 * `RippleStrokes.swift`). Authored against a 44×44 viewBox; the badge scales
 * them uniformly at draw time so they render correctly at any size.
 *
 * Indexed 1–6 from innermost to outermost, matching the stroke ids the
 * tone truth table uses.
 */
internal object RippleStrokes {
    val byId: Map<Int, Path> =
        mapOf(
            1 to stroke1(),
            2 to stroke2(),
            3 to stroke3(),
            4 to stroke4(),
            5 to stroke5(),
            6 to stroke6(),
        )

    private fun stroke1(): Path =
        Path().apply {
            moveTo(25.4147f, 30.7822f)
            cubicTo(22.7365f, 31.9714f, 19.5086f, 31.8785f, 16.4741f, 29.5504f)
            cubicTo(6.9764f, 22.2636f, 18.3687f, 7.65428f, 27.8664f, 14.941f)
            cubicTo(32.662f, 18.6202f, 32.1318f, 24.1663f, 29.2088f, 27.818f)
        }

    private fun stroke2(): Path =
        Path().apply {
            moveTo(34.1755f, 13.5272f)
            cubicTo(25.9708f, 1.34962f, 4.58761f, 9.44313f, 7.58572f, 25.087f)
        }

    private fun stroke3(): Path =
        Path().apply {
            moveTo(36.6088f, 19.5146f)
            cubicTo(39.4844f, 34.5339f, 18.0179f, 42.6778f, 10f, 31.0941f)
        }

    private fun stroke4(): Path =
        Path().apply {
            moveTo(41.458f, 26.9565f)
            cubicTo(39.2185f, 38.9628f, 23.9232f, 45.4638f, 11.4043f, 40.1005f)
        }

    private fun stroke5(): Path =
        Path().apply {
            moveTo(7.37405f, 37.1175f)
            cubicTo(-1.10595f, 29.2114f, 0.748869f, 9.24398f, 10.831f, 4.78223f)
        }

    private fun stroke6(): Path =
        Path().apply {
            moveTo(15.4023f, 2.71506f)
            cubicTo(26.241f, -0.507724f, 41.9999f, 7.38652f, 41.9999f, 21.8993f)
        }
}

/** The 44-unit source viewBox side the strokes are authored against. */
internal const val RIPPLE_VIEWBOX_SIDE = 44f
